spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -28,21 +28,48 @@
 import org.apache.iceberg.io.CloseableIterator;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.spark.source.metrics.TaskNumDeletes;
+import org.apache.iceberg.spark.source.metrics.TaskNumSplits;
 import org.apache.spark.rdd.InputFileBlockHolder;
+import org.apache.spark.sql.connector.metric.CustomTaskMetric;
+import org.apache.spark.sql.connector.read.PartitionReader;
 import org.apache.spark.sql.vectorized.ColumnarBatch;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class BatchDataReader extends BaseBatchReader<FileScanTask> {
+class BatchDataReader extends BaseBatchReader<FileScanTask>
+    implements PartitionReader<ColumnarBatch> {
+
   private static final Logger LOG = LoggerFactory.getLogger(BatchDataReader.class);
 
+  private final long numSplits;
+
+  BatchDataReader(SparkInputPartition partition, int batchSize) {
+    this(
+        partition.table(),
+        partition.taskGroup(),
+        partition.expectedSchema(),
+        partition.isCaseSensitive(),
+        batchSize);
+  }
+
   BatchDataReader(
-      ScanTaskGroup<FileScanTask> task,
```

Comment:
Most other readers have another order of parameters: table, taskGroup, expectedSchema, caseSensitive.